src/transactional-mutations-plugin/transactional-mutations-plugin.ts

```diff
@@ -0,0 +1,39 @@
+import { PolarisGraphQLContext } from '@enigmatis/polaris-common';
+import { PolarisGraphQLLogger } from '@enigmatis/polaris-graphql-logger';
+import { getPolarisConnectionManager } from '@enigmatis/polaris-typeorm';
+import { ApolloServerPlugin, GraphQLRequestContext, GraphQLRequestListener } from 'apollo-server-plugin-base';
+import { TransactionalMutationsListener } from './transactional-mutations-listener';
+import { transactionalMutationsMessages } from './transactional-mutations-messages';
+
+export class TransactionalMutationsPlugin implements ApolloServerPlugin<PolarisGraphQLContext> {
+    private readonly logger: PolarisGraphQLLogger;
+
+    constructor(logger: PolarisGraphQLLogger) {
+        this.logger = logger;
+    }
+
+    public requestDidStart(
+        requestContext: GraphQLRequestContext<PolarisGraphQLContext>,
+    ): GraphQLRequestListener<PolarisGraphQLContext> | void {
+        if (this.isTheRequestAMutation(requestContext)) {
+            this.logger.debug(transactionalMutationsMessages.pluginStartedJob, requestContext.context);
+            const queryRunner = getPolarisConnectionManager().get().manager.queryRunner;
```

Comment:
Seperate by realities

Reply:
Can we get the whole locking mechanism in a util code? I want to lock the DB too in my plugin

Reply:
can you seperate this code to a utils? I want to use it in my code as well

Reply:
what do you mean @DoctorVoid ?

Reply:
I want to use the same logic in my plugin, the lock and the rollback